benchmark/preconditioner/preconditioner.cpp

```diff
@@ -157,41 +158,32 @@ void run_preconditioner(const char *precond_name,
             for (auto i = 0u; i < FLAGS_warmup; ++i) {
                 precond->generate(system_matrix)->apply(lend(b), lend(x_clone));
             }
+            auto generate_timer = get_timer(exec, FLAGS_gpu_timer);
+            auto apply_timer = get_timer(exec, FLAGS_gpu_timer);
 
             exec->synchronize();
-            auto g_tic = std::chrono::steady_clock::now();
-
+            generate_timer->tic();
             std::unique_ptr<gko::LinOp> precond_op;
             for (auto i = 0u; i < FLAGS_repetitions; ++i) {
                 precond_op = precond->generate(system_matrix);
             }
-
-            exec->synchronize();
-            auto g_tac = std::chrono::steady_clock::now();
+            generate_timer->toc();
 
             auto generate_time =
-                std::chrono::duration_cast<std::chrono::nanoseconds>(g_tac -
-                                                                     g_tic) /
-                FLAGS_repetitions;
+                generate_timer->get_total_time() / FLAGS_repetitions;
```

Comment:
I guess you can use the `get_average_time` here as well ?

Reply:
The timer is out of the repetition loop, so `total_time = average_time` here .
To get the correct average time, need `total_time/FLAGS_repetitions` or `average_time/FLAGS_repetitions`.
`total_time` is less confusing than `average_time`?